pkg/client/cache/eventqueue.go

```diff
@@ -324,11 +339,22 @@ func (eq *EventQueue) Replace(objects []interface{}, resourceVersion string) err
 	}
 
 	if len(eq.queue) > 0 {
+		eq.lastReplaceKey = eq.queue[len(eq.queue)-1]
 		eq.cond.Broadcast()
+	} else {
+		eq.lastReplaceKey = ""
 	}
 	return nil
 }
 
+// TODO doc
+func (eq *EventQueue) LastReplaceRead() bool {
```

Comment:
> Is there a name you would suggest? I was attempting to be as literal as possible (have the items added by the last Replace() call been read from the queue?) but I'm clearly ignorant of convention here.

The intent from the consumer side is more `RelistConsumed` right?  Callers check this value to see if all the values updated in a relist have been popped.  Most callers of this function will never be calling `Replace`, so I think its doubtful that they'll know what `Replace` means in this context, but they are likely to understand this is based on listing and watching.